gen_manifest/gen_manifest.py

```diff
@@ -197,17 +197,36 @@ def get_region_tag(sample_file_path):
         region_tags.append(srt)
 
   if not region_tags:
-    sys.exit("Found no region tags.")
+    raise RegionTagError(f'Found no region tags in {sample_file_path}.')
 
   if len(region_tags) > 1:
-    sys.exit("Found too many region tags.")
+    raise RegionTagError(f'Found too many region tags in {sample_file_path}.')
 
   return region_tags[0]
 
 
-class TagNameError(Exception):
+def glob_non_yaml(pattern):
```

Comment:
Instead of ignoring just YAMLs, would it make more sense to ignore all files that does not have a valid extension for a certain `environment`?

Reply:
Hmm, that's baking language knowledge into gen-manifest, which I want to avoid. I think this is simple to begin with. If we need to be more aggressive in excluding other types of files, then we can have a separate feature addition where the user can specify explicitly the only types of file extensions to consider. How does that sound?

Reply:
We already have the information of the language in `environment`, and mapping the language to extensions should be trivial enough that can be solved by a dictionary. AFAICT all the languages we support should have just one valid extension.

Reply:
But `environment` is not required in manifest v3+. The manifest file format is very generic on purpose, and this tool should be as well. We should not bake language knowledge in here. We could specify the extensions via flags, as I suggested, but I'd rather start simple.